_includes/head.html

```diff
@@ -42,19 +42,18 @@
   {% endif %}
 
   <!-- Icons -->
-  <link rel="apple-touch-icon" sizes="57x57" href="{{ "/assets/icons/apple-touch-icon-57x57.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="114x114" href="{{ "/assets/icons/apple-touch-icon-114x114.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="72x72" href="{{ "/assets/icons/apple-touch-icon-72x72.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="144x144" href="{{ "/assets/icons/apple-touch-icon-144x144.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="60x60" href="{{ "/assets/icons/apple-touch-icon-60x60.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="120x120" href="{{ "/assets/icons/apple-touch-icon-120x120.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="76x76" href="{{ "/assets/icons/apple-touch-icon-76x76.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="152x152" href="{{ "/assets/icons/apple-touch-icon-152x152.png" | relative_url }}">
-  <link rel="apple-touch-icon" sizes="180x180" href="{{ "/assets/icons/apple-touch-icon-180x180.png" | relative_url }}">
-  <link rel="icon" type="image/png" href="{{ "/assets/icons/favicon-192x192.png" | relative_url }}" sizes="192x192">
-  <link rel="icon" type="image/png" href="{{ "/assets/icons/favicon-160x160.png" | relative_url }}" sizes="160x160">
-  <link rel="icon" type="image/png" href="{{ "/assets/icons/favicon-96x96.png" | relative_url }}" sizes="96x96">
-  <link rel="icon" type="image/png" href="{{ "/assets/icons/favicon-16x16.png" | relative_url }}" sizes="16x16">
-  <link rel="icon" type="image/png" href="{{ "/assets/icons/favicon-32x32.png" | relative_url }}" sizes="32x32">
+  {% for static_file in site.static_files -%}
+    {% if static_file.path contains "/assets/icons/apple-touch-icon-" or static_file.path contains "/assets/icons/favicon-" -%}
+      {% assign path_components = static_file.path | split: "/" -%}
+      {% assign icon = path_components[-1] | split: "-" -%}
+      {% if icon[0] == "apple" -%}
+        {% assign rel = "apple-touch-icon" -%}
+      {% else -%}
+        {% assign rel = "favicon" -%}
+      {% endif -%}
+      {% assign sizes_extension = icon[-1] | split: "." -%}
+      <link rel="{{ rel }}" type="image/png" sizes="{{ sizes_extension[0] }}" href="{{ static_file.path | relative_url }}">
+    {% endif -%}
+  {% endfor -%} 
   <link rel="shortcut icon" href="{{ "/assets/favicon.ico" | relative_url }}">
```

Comment:
There should be code that checks if this file exists before adding the `link` tag. See the [`_layouts/taxon.html`](https://github.com/legumeinfo/jekyll-theme-legumeinfo/blob/main/_layouts/taxon.html#L43) for an example.

Also, does it make sense for `favicon.ico` to be in the `assets/icons/` directory with the rest of the icons?

Reply:
> There should be code that checks if this file exists before adding the link tag. See the [_layouts/taxon.html](https://github.com/legumeinfo/jekyll-theme-legumeinfo/blob/main/_layouts/taxon.html#L43) for an example.

The `<link>` tag is in an `if` block that should accomplish that:
```
  {% for static_file in site.static_files -%}
    {% if static_file.path contains "/assets/icons/apple-touch-icon-" or static_file.path contains "/assets/icons/favicon-" -%}
    ...
      <link rel="{{ rel }}" type="image/png" sizes="{{ sizes_extension[0] }}" href="{{ static_file.path | relative_url }}">
    {% endif -%}
  {% endfor -%}
```

> Also, does it make sense for favicon.ico to be in the assets/icons/ directory with the rest of the icons?
I think so, and the README.md already mentions that favicon.ico should be in assets/icons. I've added a commit to this PR that updates the link to favicon.ico.

Reply:
> The `<link>` tag is in an `if` block that should accomplish that:
> ...
> I think so, and the README.md already mentions that favicon.ico should be in assets/icons. I've added a commit to this PR that updates the link to favicon.ico.

The tag you mention here is what I was referring to. I've confirmed it's still being added regardless of whether or not the icon file exists.

Reply:
favicon.ico is now optional, with a `<link>` generated only if it exists.